packages/router/src/util.ts

```diff
@@ -31,13 +31,23 @@ export interface ParamType {
 }
 
 /** Definitions of the core param types. */
-const coreParamTypes: Record<string, ParamType> = {
+const coreParamTypes = {
   Int: {
     constraint: /\d+/,
     transform: Number,
   },
+  Float: {
+    constraint: /[+-]?[0-9]*[.]?[0-9]+/,
```

Comment:
Should probably also match `4e8` and `1e-2`

Reply:
Mind suggesting a regex? I know you're technically correct, but realistically is this something you'd expect in a url? Perhaps we should call it Decimal instead

Reply:
>  is this something you'd expect in a url?

I never type it myself, but if it's a small enough number, JS likes to use the `e-X` notation, so I think it might end up there (cuz I'm assuming these are all generated URLs)

```
[+-]?(\d+([.]\d*)?(e[+-]?\d+)?|[.]\d+(e[+-]?\d+)?)
```

I used one of my three copy/pastes on SO to give you that. You can pay me later :D

Reply:
Updating, but that regex didn't work :/

Reply:
Updated, also added a test for this!